developer-docs-site/static/move_demo/sources/CAPMessageBoard.move

```diff
@@ -0,0 +1,202 @@
+///     (1) create its messageboard "CapMessageBoard"
+///     (2) offer participants capability to update the pinned message
+///     (3) remove the capability from a participant
+/// participant can
+///     (1) register for the board
+///     (2) redeem the offered capability to update pinned message
+///     (3) submit a new message
+///
+/// The module also emits two types of events for subscribes
+///     (1) notice cap changes event, this event contains the board address and participant offered capability
+///     (2) noice change event, this event contains the board, message and message author
+module Examples::CapMessageBoard {
+    use Std::Vector;
+    use Std::Signer;
+    use Std::Offer;
+    use Std::Event::{Self, EventHandle};
+
+    // Error map
+    const EACCOUNT_NO_NOTICE_CAP: u64 = 1;
+    const EONLY_ADMIN_CAN_REMOVE_NOTICE_CAP: u64 = 2;
+
+    struct CapMessageBoard has key {
+        notice: vector<u8>
+    }
+
+    /// provide the capability to alert the board message
+    struct NoticeChangeCapability has key, store {
+        board: address
+    }
+
+    struct NoticeCapEventHandle has key {
+        change_events: EventHandle<NoticeCapEvent>
+    }
+
+    struct NoticeCapEvent has store, drop {
+        participant: address,
+        board: address,
+    }
+
+    struct NoticeChangeEventHandle has key {
+        change_events: EventHandle<NoticeChangeEvent>
+    }
+
+    struct NoticeChangeEvent has store, drop {
+        sender: address,
+        message: vector<u8>,
+        board: address
+    }
+
+    /// init the message board and move the resource to signer
+    public fun message_board_init_internal(account: &signer) {
+        let board = CapMessageBoard{
+            notice: Vector::empty<u8>()
+        };
+        let board_addr = Signer::address_of(account);
+        move_to(account, board);
+        let notice_cap = NoticeChangeCapability{ board: board_addr };
+        move_to(account, notice_cap);
+        move_to(account, NoticeChangeEventHandle{
+            change_events: Event::new_event_handle<NoticeChangeEvent>(account)
+        });
+        move_to(account, NoticeCapEventHandle{
+            change_events: Event::new_event_handle<NoticeCapEvent>(account)
+        })
+    }
+
+    /// init a messageboard for account
+    public(script) fun message_board_init(account: signer) {
+        message_board_init_internal(&account);
+    }
+
+    public fun view_message(board_addr: address): vector<u8> acquires CapMessageBoard {
+        let board = borrow_global<CapMessageBoard>(board_addr);
+        board.notice
+    }
+
+    /// participant register to init his/her resource
+    public fun register_internal(participant: &signer) {
+        move_to(participant, NoticeChangeEventHandle{
+            change_events: Event::new_event_handle<NoticeChangeEvent>(participant)
+        })
+    }
+
+    public(script) fun register(participant: signer) {
+        register_internal(&participant);
+    }
+
```

Comment:
isn't clear to me why we want this

Reply:
events should only be emitted by the board's account imo

Reply:
participant can update the message and send the event to notify other participants that the message is updated.